cpp/farm_ng/core/logging/logger.h

```diff
@@ -14,74 +14,267 @@
 
 #pragma once
 
+#include "farm_ng/core/enum/enum.h"
 #include "farm_ng/core/logging/format.h"
 
+#include <farm_pp/preprocessor/comma.hpp>
+#include <farm_pp/preprocessor/empty.hpp>
+
 #include <filesystem>
 
+#define FARM_LEVEL_TRACE 0
+#define FARM_LEVEL_DEBUG 1
+#define FARM_LEVEL_INFO 2
+#define FARM_LEVEL_WARN 3
+#define FARM_LEVEL_ERROR 4
+#define FARM_LEVEL_CRITICAL 5
+#define FARM_LEVEL_OFF 6
+
+// Define FARM_LOG_LEVEL to set the compile-time log level
+#ifndef FARM_LOG_LEVEL
+#define FARM_LOG_LEVEL FARM_LEVEL_DEBUG
```

Comment:
Does this mean we are defaulting to FARM_LEVEL_DEBUG? Should't we rather default to FARM_LEVEL_WARN, and then enable per compilation unit if we need more?

Reply:
changed default to INFO (as that's what our current behavior was)

Reply:
and that's the default used by other libraries like `spdlog`